packages/components/src/entities/FindSamplesByIdsPageBase.tsx

```diff
@@ -203,8 +203,14 @@ const FindSamplesByIdsPageBaseImpl: FC<Props> = memo(props => {
         if (!sampleListModelId) return;
 
         const model = queryModels[sampleListModelId];
-        if (model && !model.isLoading && model.rowCount > 0) actions.selectAllRows(sampleListModelId);
-    }, [sampleListModelId, queryModels[sampleListModelId]?.isLoading]);
+        if (model && !model.isLoading && !model.isLoadingTotalCount && model.rowCount > 0) {
+            actions.selectAllRows(sampleListModelId);
+        }
+    }, [
+        sampleListModelId,
+        queryModels[sampleListModelId]?.isLoading,
+        queryModels[sampleListModelId]?.isLoadingTotalCount,
```

Comment:
In this case we just want to execute the useEffect when the loading state is done so that after the totalRowCount is finished loading we select all rows in the grid, just once.

Reply:
Worth a comment in the code?

Reply:
done